launch/launch/frontend/parser.py

```diff
@@ -132,12 +140,56 @@ def get_parser_from_extension(
         extension: Text,
     ) -> Optional[Type['Parser']]:
         """Return an entity loaded with a markup file."""
+        warnings.warn(
+            'Parser.get_parser_from_extension is deprecated, '
+            'use Parser.get_parsers_from_filename instead')
         cls.load_parser_implementations()
         try:
             return cls.frontend_parsers[extension]
         except KeyError:
             raise RuntimeError('Not recognized frontend implementation')
 
+    @classmethod
+    def may_parse(
+        cls,
+        filename: Text,
+    ) -> bool:
+        """Return `true` if the filename is valid for this parser."""
```

Comment:
@christophebedard nit:
```suggestion
        """Return `True` if the filename is valid for this parser."""
```

Reply:
fixed in db9f1f2e9fce7b2268daa441bf13b0f9346206ac